NDTensors/src/UnallocatedArrays/src/unallocatedzeros.jl

```diff
@@ -0,0 +1,143 @@
+struct UnallocatedZeros{ElT,N,Axes,Alloc<:AbstractArray{ElT,N}} <:
+       FillArrays.AbstractZeros{ElT,N,Axes}
+  z::FillArrays.Zeros{ElT,N,Axes}
+  function UnallocatedZeros{ElT,N,Alloc}(inds::Tuple) where {ElT,N,Alloc}
+    z = FillArrays.Zeros{ElT,N}(inds)
+    Axes = typeof(FillArrays.axes(z))
+    return new{ElT,N,Axes,Alloc}(z)
+  end
+
+  function UnallocatedZeros{ElT,N,Alloc}(::Tuple{}) where {ElT,N,Alloc}
+    @assert N == 1
+    z = FillArrays.Zeros{ElT,N}(1)
+    Axes = typeof(FillArrays.axes(z))
+    return new{ElT,N,Axes,Alloc}(z)
+  end
+
+  function UnallocatedZeros{ElT,N,Axes,Alloc}(inds::Tuple) where {ElT,N,Axes,Alloc}
+    @assert Axes == typeof(Base.axes(inds))
+    z = FillArrays.Zeros{ElT,N}(inds)
+    return new{ElT,N,Axes,Alloc}(z)
+  end
+end
+
+function UnallocatedZeros{ElT,N,Axes,Alloc}() where {ElT,N,Axes,Alloc<:AbstractArray{ElT,N}}
+  return UnallocatedZeros{ElT,N,Axes,Alloc}[]
+end
```

Comment:
I'm getting a bit lost with all of these constructors.

Can we try to design them in a similar way to how the constructors for the `Zeros` type is defined in `FillArrays`?